### PR TITLE
implement decimalToPrecision everywhere

### PR DESCRIPTION
**Pull request by frosty00**
I tried to keep the behavior of the all the methods the same. Hopefully this fixes bugs in the `priceToPrecision` and related functions called by create order. `toFixed` and f'{:.nf}' both round so I kept the `ROUND` behaviour while `this.truncate` is replaced by `TRUNCATE`. 

I dislike that `amountToPrecision` returns a float while all other methods return a string but I wanted to be consistent with what it used to return so I kept it.

### CODE REVIEW

**Comment by frosty00** [comment]
@kroitor could you take a look at the last commit. Basically I was annoyed that `amountToPrecision` returned a float, despite it being used in 90% of the cases where `amountToString` should have been used (createOrder requests etc...), so now `amountToString` no longer exists. Moreover, to change the counting mode, `precisionMode` can be set instead of subclassing all the methods.

**Comment by kroitor** [comment]
As described in #3800:

`decimal_to_precision(0.0000001, TRUNCATE, 8, DECIMAL_PLACES) = '9.99999999'`

I've added a test for it into this branch.

```Python
>>> import ccxt
>>> x = ccxt.binance()
>>> x.decimal_to_precision(0.0000001, ccxt.TRUNCATE, 8, ccxt.DECIMAL_PLACES)
'9.99999999'
>>> x.decimal_to_precision('0.0000001', ccxt.TRUNCATE, 8, ccxt.DECIMAL_PLACES)
'1E-7'
>>> x.decimal_to_precision('0.00000001', ccxt.TRUNCATE, 8, ccxt.DECIMAL_PLACES)
'1E-8'
>>> x.decimal_to_precision(0.00000001, ccxt.TRUNCATE, 8, ccxt.DECIMAL_PLACES)
'1'
>>>
```

**Comment by frosty00** [comment]
#3800

File: python/ccxt/base/decimal_to_precision.py [hunk 1]
```diff
@@ -40,16 +40,16 @@ def decimal_to_precision(n, rounding_mode=ROUND, precision=None, counting_mode=D
...
+    string = '{:f}'.format(dec)  # convert to string using .format to avoid engineering notation
     precise = None
 
     def power_of_10(x):
         return decimal.Decimal('10') ** (-x)
 
     if rounding_mode == ROUND:
         if counting_mode == DECIMAL_PLACES:
-            precise = str(dec.quantize(power_of_10(precision)))  # ROUND_HALF_EVEN is default context
+            precise = '{:f}'.format(dec.quantize(power_of_10(precision)))  # ROUND_HALF_EVEN is default context
```
Comment (beniloew) [inline python/ccxt/base/decimal_to_precision.py]: Did you test this with a number like 3456.00004?
I think that the quantize needs the number of digits of the whole number, not just after the floating point

Comment (beniloew) [reply python/ccxt/base/decimal_to_precision.py]: I did test it in my patch with the truncate mode, but did not in round node

Comment (frosty00) [reply python/ccxt/base/decimal_to_precision.py]: > Did you test this with a number like 3456.00004?

```
decimal_to_precision(3456.00004, TRUNCATE, 5, DECIMAL_PLACES)
Out[7]: '3456.00004'
decimal_to_precision(3456.00004, TRUNCATE, 4, DECIMAL_PLACES)
Out[8]: '3456'
```

We have done a lot of testing: https://github.com/ccxt/ccxt/blob/master/js/test/base/functions/test.number.js

**Comment by frosty00** [comment]
all xToPrecision's should be removed from subclasses and AFAIK only `this.precisionMode` needs to be overidden (e.g. bitfinex and sigfigs)

**Pull request merged**